eng/pipelines/templates/steps/sync-repo.yml

```diff
@@ -0,0 +1,29 @@
+parameters:
```

Comment:
Is there a reason you are creating a new one here instead of updating the existing one https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/steps/sync-repo-branch.yml?

Reply:
This is an improvement of mirror-files, for better isolation testing. https://github.com/Azure/azure-sdk-tools/pull/5479#issuecomment-1433861143. There are functionally consistent for two files.Keeping them will cause confusion, I will Delete sync-repo-branch.yml file and update mirror-files.yml.

Reply:
I'm expecting an updated sync-repo-branch.yml as opposed to a new sync-repo.yml file. Is there a reason you are creating an additional one instead of updating the existing one?